packages/volto/src/components/manage/Controlpanels/Users/UsersControlpanel.jsx

```diff
@@ -430,359 +431,303 @@ class UsersControlpanel extends Component {
-      adduserschema = JSON.parse(
-        JSON.stringify(this.props?.userschema?.userschema),
-      );
-      adduserschema.properties['username'] = {
-        title: this.props.intl.formatMessage(messages.addUserFormUsernameTitle),
-        type: 'string',
-        description: this.props.intl.formatMessage(
-          messages.addUserFormUsernameDescription,
-        ),
-      };
-      adduserschema.properties['password'] = {
-        title: this.props.intl.formatMessage(messages.addUserFormPasswordTitle),
-        type: 'password',
-        description: this.props.intl.formatMessage(
-          messages.addUserFormPasswordDescription,
-        ),
-        widget: 'password',
-      };
-      adduserschema.properties['sendPasswordReset'] = {
-        title: this.props.intl.formatMessage(
-          messages.addUserFormSendPasswordResetTitle,
-        ),
-        type: 'boolean',
-      };
-      adduserschema.properties['roles'] = {
-        title: this.props.intl.formatMessage(messages.addUserFormRolesTitle),
-        type: 'array',
-        choices: this.canAssignAdd(isUserManager).map((role) => [
-          role.id,
-          role.title,
-        ]),
-        noValueOption: false,
-      };
-      adduserschema.properties['groups'] = {
-        title: this.props.intl.formatMessage(messages.addUserGroupNameTitle),
-        type: 'array',
-        choices: this.props.groups
-          .filter((group) => canAssignGroup(isUserManager, group))
-          .map((group) => [group.id, group.id]),
-        noValueOption: false,
-      };
-      if (
-        adduserschema.fieldsets &&
-        adduserschema.fieldsets.length > 0 &&
-        !adduserschema.fieldsets[0]['fields'].includes('username')
-      ) {
-        adduserschema.fieldsets[0]['fields'] = adduserschema.fieldsets[0][
-          'fields'
-        ].concat([
+  }, [createRequest?.error, createRequest?.loading, onAddUserError]);
+
+  useEffect(() => {
+    if (loadRolesRequest?.error && !loadRolesRequest?.loading) {
+      setError(loadRolesRequest.error);
+    }
+  }, [loadRolesRequest?.error, loadRolesRequest?.loading]);
+
+  if (error) {
+    return <Error error={error} />;
+  }
+
+  const usernameToDelete = userToDelete ? userToDelete.username : '';
+
+  // Copy the userschema using JSON serialization/deserialization
+  // this is really ugly, but if we don't do this the original value
+  // of the userschema is changed and it is used like that through
+  // the lifecycle of the application
+  let adduserschema = {};
+  let isUserManager = false;
+  if (userschema?.loaded) {
+    isUserManager = isManager(user);
+    adduserschema = JSON.parse(JSON.stringify(userschema?.userschema));
+
+    // Add custom form fields to the schema
+    adduserschema.properties.username = {
+      title: intl.formatMessage(messages.addUserFormUsernameTitle),
+      type: 'string',
+      description: intl.formatMessage(messages.addUserFormUsernameDescription),
+    };
```

Comment:
Field `username` should not be added if `loginUsingEmail` is true.

Reply:
This is a problem that occurs in Volto today. See:

https://github.com/plone/volto/issues/5951

In this PR I'm only concerned with migrating to a functional component. This already changes a lot of things. Fixing a bug in mine would make the PR more confusing. This can be fixed in another PR, after the component has been migrated.